src/Inventory/Asset/NetworkPort.php

```diff
@@ -250,8 +250,25 @@ private function handleLLDPConnection(\stdClass $port, int $netports_id)
             $rule->getCollectionPart();
             $rule->processAllRules($input, [], ['class' => $this]);
 
-            if (count($this->connection_ports)) {
-                $connections_id = current(current($this->connection_ports));
+            if (count($this->connection_ports) != 1) {
+                continue;
+            }
+
+            $connection_ports = current($this->connection_ports);
+            if (count($connection_ports) == 1) { // single NetworkPort
+                $connections_id = current($connection_ports);
+            } else { // multiple NetworkPorts
+                $networkPort = new \NetworkPort();
+                foreach (array_keys($connection_ports) as $k) {
+                    $networkPort->getFromDB($k);
+                    if ($networkPort->fields['logical_number'] > 0) {
+                        $connections_id = $k;
```

Comment:
Should a call to `$this->addPortsWiring($netports_id, $k);` be made for each port having a logical number > 0?

Reply:
No, LLDP neighbors are connected directly in 99.999% of cases, so only one real port should be connected. There are some potential cases when LLDP may report multiple neighbors on a port, but i've seen it only once when port mirroring feature was turned on and i consider this a misconfiguration.

Reply:
@cedric-anne take a look at [c6d95dc](https://github.com/glpi-project/glpi/commit/c6d95dca189f91e8fe264835c67fb890bd92bbcb)
I added a code to find real port info from LLDP sysmac and local name, and I'd like to hear ideas for a more generic solution which would also handle CDP-provided ip and ifdescr.

All of this to reduce the chance of getting multiple ports.

Reply:
Never mind, lldp import works, cdp import works. No need to complicate.